Added  a anime trailers page

soulcramer opened the pull request.

soulcramer committed:
✨ Added anime trailers page.

soulcramer committed:
💄 Show trailers from other source than only Youtube on anime page

soulcramer committed:
💄 Display the trailers with a better visual and responsive.

akyoto commented:
I believe the Youtube restriction was in place to discourage non-Youtube sources because we wanted to store them locally as WebM files at some point.

soulcramer committed:
👌🐛 Revert commit 201e18d and display the latest valid trailer.

After reverting to the previous state of the anime page, I noticed that if the latest trailer is not "valid" (currently which is not coming from Youtube and doesn't have a serviceId) we didn't check if there was a valid one.
To resolve this issue I've added a trailer parameter to the Anime component which should be the latest valid trailer that we have. Otherwise, we don't display the trailer component.
